docs(arby): add arby's logic documentation

erkarl opened the pull request:
I started off with trying to document everything in text format, but ended up with a flowchart that attempts to describe arby's business logic (arbitrage trade) in as much detail as possible. Please keep in mind that this is not 100% how Arby currently functions, rather an ideal flow of things.

Things to add/discuss:
- fail safe handling (when we lose connection to OpenDEX or CentralizedExchange)
- maximum buy/sell amount calculation (can probably be documented in text form better)
- trade stages (started/stopped/updatingOrders/waitingOpenDEX/waitingCentralizedExchange)
- logic that handles queue for arbitrage trades when multiple trading pairs are active (we only want to process 1 trade at a time)
- OpenDEX taker mode

Commit:
docs(arby): add arby's logic documentation

kilrau commented on docs/arby/README.md:
```diff
@@ -1,4 +1,10 @@
 # Arby
 Arby looks for arbitrage opportunities on the OpenDEX network. Once a profitable trade appears it will automatically attempt to enter it. It will also enter active orders to the orderbook and updates them every 1 minute by default.
 
+Currently, it is possible to use Arby for arbitrage trades between OpenDEX and Binance for LTCBTC trading pair. The system is designed in a way that allows more exchanges and trading pairs to be added in the future.
```
```suggestion
Currently, it is possible to use Arby for arbitrage trades between OpenDEX and Binance for the LTC/BTC trading pair. The system is designed in a way that allows more exchanges and trading pairs to be added in the future.
```

kilrau commented on docs/arby/README.md:
```diff
@@ -1,4 +1,10 @@
+
 [Setup Guide](./setup.md)
+
+## How does it work?
```
Diagram comments:
- OpenDEX Taker (TBD) will be the Hedgy functionality I take it?
- `VOLATILITY_TIMEOUT` is https://github.com/ExchangeUnion/market-maker-tools/issues/4, right? Already implemented?
- market buy/sell order fallback: I think we should never do market orders, rather issue a new limit order if not filled within `LIMIT_ORDER_TIMEOUT`

erkarl replied:
> OpenDEX Taker (TBD) will be the Hedgy functionality I take it?

Not exactly, hedgy will rebalance regardless of the price. This is for the scenario where there is a better price on OpenDEX (highly unlikely in the short term) than on centralized exchange.

> market buy/sell order fallback: I think we should never do market orders, rather issue a new limit order if not filled within LIMIT_ORDER_TIMEOUT

If the market becomes volatile while waiting for the asymmetric trade it could cause huge losses. What happens if the second limit order does not succeed? Third? Forth? Not sure if 100% excluding market orders is the way to go.

erkarl replied:
> VOLATILITY_TIMEOUT is #4, right? Already implemented?

Not yet (as mentioned in the OP this describes the ideal logic and does not reflect the current logic 100%). 

I wanted to clarify what the logic should be when "returning" from the break/timeout period. What is the criteria to ensure that the market is no longer volatile? Price movement X% in Y% of time?

kilrau replied:
> If the market becomes volatile while waiting for the asymmetric trade it could cause huge losses. What happens if the second limit order does not succeed? Third? Forth? Not sure if 100% excluding market orders is the way to go.

Moved this to a separate issue: https://github.com/ExchangeUnion/market-maker-tools/issues/7

kilrau replied:
> > VOLATILITY_TIMEOUT is #4, right? Already implemented?
> 
> Not yet (as mentioned in the OP this describes the ideal logic and does not reflect the current logic 100%).
> 
> I wanted to clarify what the logic should be when "returning" from the break/timeout period. What is the criteria to ensure that the market is no longer volatile? Price movement X% in Y% of time?

Also moved to separate issue: https://github.com/ExchangeUnion/market-maker-tools/issues/8

xVETTEx commented:
For me it would be easier to read flowchart if it would go more from top to bottom something like this. 
![arby_upgraded](https://user-images.githubusercontent.com/46370333/77149851-3632de80-6a9b-11ea-8842-15002420521e.png)
Formula for order's price is still missing, it could be in text format.

Now there is "Marker sell/buy order" , I assume it means marker means market? My opinion is that arby shouldn't make limit order to binance or other exchanges at all, money is losed much when price go wrong direction and arby will make market order with much worse price than order in OpeDEX was. I can show some numbers from binance to demonstrate why IMO limit order is not good idea, if needed.

Now chart ends to "trade complete", but arby don't stop running there, so maybe some arrow to somwhere up from "trade compete"?

Commit:
fixup! docs(arby): add arby's logic documentation

erkarl commented:
> For me it would be easier to read flowchart if it would go more from top to bottom something like this.

Thanks for cleaning up the flowchart. Updated and also added the latest dia file.

> Now there is "Marker sell/buy order" , I assume it means marker means market? My opinion is that arby shouldn't make limit order to binance or other exchanges at all, money is losed much when price go wrong direction and arby will make market order with much worse price than order in OpeDEX was. I can show some numbers from binance to demonstrate why IMO limit order is not good idea, if needed.

I'd be very interested in seeing the numbers.

> Now chart ends to "trade complete", but arby don't stop running there, so maybe some arrow to somwhere up from "trade compete"?

Good point. Adding an arrow to restart the flow after trade complete.
